### ISSUE TITLE
As view fields, titles for Pageless nodes show links even for those without access.

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#5214]
**Description of the bug**

This is a follow-up to https://github.com/backdrop/backdrop-issues/issues/5181

> with Views lists, node titles still show as hyperlinks when using fields

I looked into this, but in the view we don't currently have the node type, or the rendered node to pass into `node_type_get_type()`. We need the type in order to find out if the node has a hidden path. 

I'm hesitant to load a node where we didn't load one before (especially in views where there are likely to be hundreds of node) due to performance implications, so I'd rather we find a way to pass along the node type as an `additional_field` much like we pass along `nid` as an `additional_field`. I don't think this will affect performance since we can pull it out of the node table along with `nid`.

**Steps To Reproduce**

To reproduce the behavior:
1. Change the Post content type to be pageless.
2. Build a view of fields, leave the box checked to link the title to node.
3. View the view as authenticated (the title to Post nodes should be linked).
4. View the view as anonymous (the title to Post nodes should NOT be linked, as the link goes to 404).

**Actual behavior**

Links to 404s are generated since node module does not check for hidden paths before generating this link in views.

**Expected behavior**

The link should appear for those with access to "view hidden paths" but the title should not be linked for those who do not have this access.

**Comment by herbdool** [comment]
I don't think this is a bug in Views. We can't expect Views, in my opinion, to do things like this. If we have a hidden feature that removes links depending on access control on titles, then why wouldn't we do it on *all* links appearing in Views? There are other link fields in Views too.

If there's anything here, it's better to make it an obvious option to apply access control to links.

**Comment by jenlampton** [comment]
The node title handler is provided by node module, so it's a bug in Node. I
think it's reasonable for node module to know about its own features, so
I'm planning on trying the change there.

I wasn't planning on adding a wider access control feature to all views
links, but that's also a good idea!

**Comment by herbdool** [comment]
This is my understanding. When adding a title field to a View, you have this option: "Link this field..."

![Screenshot from 2021-09-13 13-24-47](https://user-images.githubusercontent.com/1071483/133129196-64c2fa2a-58cf-4024-872b-0f1987149f4e.png)

This also appears in the field: `NID` and perhaps others. And there is also `Content: Link` which doesn't have an option, it'll just output a link to the node.

Anyway, my understanding of this issue is that regardless of the selection of the option "Link this field..." it wants to override that with access control. If so, this isn't really a bug, but a new feature. And it's not specific to just "pageless nodes" but to any access control mechanism, is it not?

**Comment by jenlampton** [comment]
The wider access control change (preventing 503) could be a new feature. The intended behavior for node title was simply to remove the link when the resulting page would be a 404, same as we did for the node title on teasers.
